File: 2022/02_rock_paper_scissors/part2.py
# A = ROCK 1
# B = PAPER 2
# C = SCISSORS 3

# X = LOSE 0
# Y = DRAW 3
# Z = WIN 6

A_X = 0 + 3
A_Y = 3 + 1
A_Z = 6 + 2
B_X = 0 + 1
B_Y = 3 + 2
B_Z = 6 + 3
C_X = 0 + 2
C_Y = 3 + 3
C_Z = 6 + 1

with open("input.txt", "r") as file:
    sum = 0
    for line in file:
        if line.strip() == "A X":
            sum += A_X
        elif line.strip() == "A Y":
            sum += A_Y
        elif line.strip() == "A Z":
            sum += A_Z
        elif line.strip() == "B X":
            sum += B_X
        elif line.strip() == "B Y":
            sum += B_Y
        elif line.strip() == "B Z":
            sum += B_Z
        elif line.strip() == "C X":
            sum += C_X
        elif line.strip() == "C Y":
            sum += C_Y
        elif line.strip() == "C Z":
            sum += C_Z
        else:
            print(line)
    print(sum)

File: 2022/02_rock_paper_scissors/part2.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer input.Close()

	A_X := 0 + 3
	A_Y := 3 + 1
	A_Z := 6 + 2
	B_X := 0 + 1
	B_Y := 3 + 2
	B_Z := 6 + 3
	C_X := 0 + 2
	C_Y := 3 + 3
	C_Z := 6 + 1

	sum := 0

	scanner := bufio.NewScanner(input)
	for scanner.Scan() {
		line := scanner.Text()

		if line == "A X" {
			sum += A_X
		} else if line == "A Y" {
			sum += A_Y
		} else if line == "A Z" {
			sum += A_Z
		} else if line == "B X" {
			sum += B_X
		} else if line == "B Y" {
			sum += B_Y
		} else if line == "B Z" {
			sum += B_Z
		} else if line == "C X" {
			sum += C_X
		} else if line == "C Y" {
			sum += C_Y
		} else if line == "C Z" {
			sum += C_Z
		} else {
			fmt.Println(line)
		}
	}

	fmt.Println(sum)

	if err := scanner.Err(); err != nil {
		fmt.Println(err)
	}
}
